liota/dccs/iotcc.py

```diff
@@ -75,21 +77,21 @@ def on_receive_safe(msg):
             try:
                 log.debug("Received msg: {0}".format(msg))
                 json_msg = json.loads(msg)
-                self.proto.on_receive(json.loads(msg))
-                log.debug("Processed msg: {0}".format(json_msg["type"]))
-                if json_msg["type"] == "connection_verified":
+                self.proto.on_receive(json_msg)
+                if json_msg["type"] == "connection_response" and json_msg["body"]["result"] == "succeeded":
                     log.info("Connection verified")
-                    exit()
-            except Exception:
-                raise
-                log.exception(
-                    "Error received on connecting to DCC instance. Please verify the credentials and try again.")
-
-        thread = threading.Thread(target=self.con.run)
-        self.con.on_receive = on_receive_safe
+                    return True
+                else:
+                    log.debug("Processed msg: {0}".format(json_msg["type"]))
+                    on_receive_safe(self.recv_msg_queue.get())
+            except Exception as error:
+                log.error("HelixProtocolException: " + repr(error))
+
+        thread = threading.Thread(target=self.comms.receive)
```

Comment:
We need to change Websocket client to WebSocket app to achieve his functionality, marked as an issue.